packages/dx-react-grid/docs/guides/custom-plugin-development.md

```diff
@@ -0,0 +1,702 @@
+      >
+        <Template name="toolbarContent">
+          <TemplatePlaceholder />
+          <TemplateConnector>
+          {({
+            toolbarFilterValue,
+            ...
+            }) => (
+              <Select
+                value={toolbarFilterValue}
+              >
+                ...
+              </Select>
+          )}
+          </TemplateConnector>
+        </Template>
+      </Plugin>
+    );
+  }
+}
+```
+
+## Use the Functionality of Built-In Plugins
+
+We need to determine when a value is changed and filter the grid based on the value. We store the `filterValue` property in state object of the `ToolbarFilterState` plugin. However, the UI element that affects this value is rendered by the `ToolbarFilter` plugin. How can we detect changes in one plugin and change the required value in the state object of another plugin? We need to use an [action](../../../core/docs/reference/action.md). Let's name it `changeToolbarFilterValue`.
+
+*file toolbar-filter-state.js:*
+```jsx
+...
+import {
+    ..., Action
+} from '@devexpress/dx-react-core';
+
+export class ToolbarFilterState extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    ...
+    this.state = {
+      filterValue: this.props.defaultFilterValue
+    };
+    this.changeValue = this.changeValue.bind(this);
+  }
+
+  changeValue() {
+  }
+
+  render() {
+    return (
+      <Plugin
+        name="ToolbarFilterState"
+      >
+        <Action name="changeToolbarFilterValue" action={this.changeValue} />
+      </Plugin>
+    );
+  }
+}
+```
+Now, `changeToolbarFilterValue` is available in any plugin that is used with `ToolbarFilterState`. In our scenario, we may get it using [TemplateConnector](../../../core/docs/reference/template-connector.md) in the `ToolbarFilter` plugin. The following code snippet shows how to get the `changeToolbarFilterValue` action and use it as the `Select.onChange` event handler.
+
+*toolbar-filter.js:*
+```jsx
+...
+export class ToolbarFilter extends React.PureComponent {
+  render() {
+    return (
+      <Plugin
+        ...
+      >
+        <Template name="toolbarContent">
+          <TemplatePlaceholder />
+          <TemplateConnector>
+          {({
+            ... //Accessing getters
+           }, {
+            changeToolbarFilterValue //Accessing actions
+          }) => (
+            <Select
+              ...
+              onChange={changeToolbarFilterValue}
+            >
+              ...
+            </Select>
+          )}
+          </TemplateConnector>
+        </Template>
+      </Plugin>
+    );
+  }
+}
+```
+
+The event handler passes the `event` parameter. We can get a new value using this parameter and pass it to the state object of the `ToolbarFilterState` plugin.
+
+*file toolbar-filter-state.js:*
+```jsx
+...
+export class ToolbarFilterState extends React.PureComponent {
+  ...
+  changeValue(event) {
+    this.setState({
+      filterValue: event.target.value
```

Comment:
A state plugin shouldn't know anything about the UI specifics. So, the `changeValue` action should accept a ready-to-use filter value, not a UI event.